resource/graph_node.go

```diff
@@ -11,6 +11,24 @@ import (
 	"go.viam.com/rdk/logging"
 )
 
+// NodeState captures the configuration lifecycle state of a resource node.
+type NodeState uint8
+
+const (
+	// NodeStateUnconfigured denotes a newly created resource.
+	NodeStateUnconfigured NodeState = iota
```

Comment:
prefer if we just kept it 1-1 to proto - so Initialize, Configure, Ready, Remove, Unhealthy.

Reply:
Oh, I was planning to make these the proto states as well. I think `Unconfigured`, `Configuring`, `Removing` are more "stateful" descriptors. The original ones sound more like commands.

Reply:
Updated related tickets with a comment: https://viam.atlassian.net/browse/RSDK-7900?focusedCommentId=31236, https://viam.atlassian.net/browse/RSDK-7899

Let me know if you disagree

Reply:
i think the name changes are fine, just surprised they changed lol

Reply:
yeah, naming stuff is hard :name_badge: